src/mongoose_account_api.erl

```diff
@@ -0,0 +1,327 @@
+            {user_does_not_exist, io_lib:format("User ~s does not exist", [jid:to_binary(JID)])}
+    end.
+
+-spec check_password(jid:user(), jid:server(), binary()) -> check_password_result().
+check_password(User, Host, Password) ->
+    JID = jid:make(User, Host, <<>>),
+    check_password(JID, Password).
+
+-spec check_password(jid:jid(), binary()) -> check_password_result().
+check_password(JID, Password) ->
+    case ejabberd_auth:does_user_exist(JID) of
+        true ->
+            case ejabberd_auth:check_password(JID, Password) of
+                true ->
+                    {ok, io_lib:format("Password '~s' for user ~s is correct",
+                                       [Password, jid:to_binary(JID)])};
+                false ->
+                    {incorrect, io_lib:format("Password '~s' for user ~s is incorrect",
+                                              [Password, jid:to_binary(JID)])}
+            end;
+        false ->
+            {user_does_not_exist,
+            io_lib:format("Password ~s for user ~s is incorrect because this user does not"
+                          " exist", [Password, jid:to_binary(JID)])}
+    end.
+
+-spec check_password_hash(jid:user(), jid:server(), string(), string()) ->
+    check_password_hash_result().
+check_password_hash(User, Host, PasswordHash, HashMethod) ->
+    JID = jid:make(User, Host, <<>>),
+    check_password_hash(JID, PasswordHash, HashMethod).
+
+-spec check_password_hash(jid:jid(), string(), string()) -> check_password_hash_result().
+check_password_hash(JID, PasswordHash, HashMethod) ->
+    AccountPass = ejabberd_auth:get_password_s(JID),
+    AccountPassHash = case HashMethod of
+        "md5" -> get_md5(AccountPass);
+        "sha" -> get_sha(AccountPass);
+        _ -> undefined
+    end,
+    case AccountPassHash of
+        undefined ->
+            {error, "Hash for password is undefined"};
+        PasswordHash ->
+            {ok, "Password hash is correct"};
+        _->
+            {incorrect, "Password hash is incorrect"}
+    end.
+
+-spec num_active_users(jid:lserver(), integer()) -> num_active_users_result().
+num_active_users(Domain, Days) ->
+    TimeStamp = erlang:system_time(second),
+    TS = TimeStamp - Days * 86400,
+    try
+        {ok, HostType} = mongoose_domain_api:get_domain_host_type(Domain),
+        Num = mod_last:count_active_users(HostType, Domain, TS),
+        {ok, Num}
+    catch
+        _:_ ->
+            Message = io_lib:format("Cannot count active users for domain ~s", [Domain]),
+            {cannot_count, Message}
+    end.
+
+-spec ban_account(jid:user(), jid:server(), binary() | string()) ->
+    {ok, string()} | {error, string()}.
+ban_account(User, Host, ReasonText) ->
+    JID = jid:make(User, Host, <<>>),
+    ban_account(JID, ReasonText).
+
+-spec ban_account(jid:jid(), binary() | string()) -> change_password_result().
+ban_account(JID, ReasonText) ->
+    Reason = service_admin_extra_sessions:prepare_reason(ReasonText),
```

Comment:
This needs to be moved to `mongoose_session_api` - otherwise I think we may forget about such dependencies that we need to avoid.